### PR TITLE
[ChangeLog] Add entries for SE-0333, -0334, and -0349

### PR DESCRIPTION
**Pull request by glessard**
Add ChangeLog entries for SE-0333, -0334, and -0349

### CODE REVIEW

**Comment by glessard** [comment]
@swift-ci please smoke test

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,72 @@ _**Note:** This is in reverse chronological order, so newer entries are added to
...
+  - `UnsafeRawPointer` and `UnsafeMutableRawPointer` have new functionality for
+  pointer arithmetic, adding functions to obtain a pointer advanced to the next
+  or previous alignment boundary:
+
+    ```swift
+    extension UnsafeRawPointer {
+      public func alignedUp<T>(for: T.type) -> UnsafeRawPointer
+      public func alignedDown<T>(for: T.type) -> UnsafeRawPointer
+      public func alignedUp(toMultipleOf alignment: Int) -> UnsafeRawPointer
+    	public func alignedDown(toMultipleOf alignment: Int) -> UnsafeRawPointer
```
Comment (benrimmington) [inline CHANGELOG.md:43]: Tab → spaces

```suggestion
      public func alignedDown(toMultipleOf alignment: Int) -> UnsafeRawPointer
```

Comment (glessard) [reply CHANGELOG.md]: Thanks @ben

**Comment by glessard** [comment]
@swift-ci please smoke test and merge

**Pull request merged**